File: kitchenassistant-infra/src/main/kotlin/pl/szczeliniak/kitchenassistant/swagger/SwaggerConfiguration.java
package pl.szczeliniak.kitchenassistant.swagger;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import springfox.documentation.builders.PathSelectors;
import springfox.documentation.builders.RequestHandlerSelectors;
import springfox.documentation.service.*;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.spi.service.contexts.SecurityContext;
import springfox.documentation.spring.web.plugins.Docket;
import springfox.documentation.swagger2.annotations.EnableSwagger2;

import java.util.Collections;
import java.util.List;

@Configuration
@EnableSwagger2
public class SwaggerConfiguration {

    @Value("${application.name}")
    private String applicationName;
    @Value("${application.description}")
    private String applicationDescription;
    @Value("${application.version}")
    private String applicationVersion;
    @Value("${application.creator.name}")
    private String applicationCreatorName;
    @Value("${application.creator.url}")
    private String applicationCreatorUrl;
    @Value("${application.creator.email}")
    private String applicationCreatorEmail;

    @Bean
    public Docket api() {
        Docket docket = new Docket(DocumentationType.SWAGGER_2);
        return docket.apiInfo(apiInfo())
                .securityContexts(Collections.singletonList(securityContext()))
                .securitySchemes(Collections.singletonList(apiKey()))
                .select()
                .apis(RequestHandlerSelectors.basePackage("pl.szczeliniak.todoapp"))
                .paths(PathSelectors.any())
                .build();
    }

    private ApiKey apiKey() {
        return new ApiKey("JWT", "Authorization", "header");
    }

    private SecurityContext securityContext() {
        return SecurityContext.builder().securityReferences(defaultAuth()).build();
    }

    private List<SecurityReference> defaultAuth() {
        AuthorizationScope authorizationScope = new AuthorizationScope("global", "accessEverything");
        AuthorizationScope[] authorizationScopes = new AuthorizationScope[1];
        authorizationScopes[0] = authorizationScope;
        return List.of(new SecurityReference("JWT", authorizationScopes));
    }

    private ApiInfo apiInfo() {
        return new ApiInfo(
                applicationName,
                applicationDescription,
                applicationVersion,
                null,
                new Contact(applicationCreatorName, applicationCreatorUrl, applicationCreatorEmail),
                null,
                null,
                Collections.emptyList());
    }

}

File: kitchenassistant-infra/src/main/kotlin/pl/szczeliniak/kitchenassistant/swagger/SwaggerConfiguration.kt
package pl.szczeliniak.kitchenassistant.swagger

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import springfox.documentation.builders.PathSelectors
import springfox.documentation.builders.RequestHandlerSelectors
import springfox.documentation.service.*
import springfox.documentation.spi.DocumentationType
import springfox.documentation.spi.service.contexts.SecurityContext
import springfox.documentation.spring.web.plugins.Docket
import springfox.documentation.swagger2.annotations.EnableSwagger2

@Configuration
@EnableSwagger2
class SwaggerConfiguration(
    @Value("\${application.name}")
    private val applicationName: String,
    @Value("\${application.description}")
    private val applicationDescription: String,
    @Value("\${application.version}")
    private val applicationVersion: String,
    @Value("\${application.creator.name}")
    private val applicationCreatorName: String,
    @Value("\${application.creator.url}")
    private val applicationCreatorUrl: String,
    @Value("\${application.creator.email}")
    private val applicationCreatorEmail: String
) {

    @Bean
    fun api(): Docket {
        return Docket(DocumentationType.SWAGGER_2).apiInfo(apiInfo())
            .securityContexts(listOf(securityContext()))
            .securitySchemes(listOf(apiKey()))
            .select()
            .apis(RequestHandlerSelectors.basePackage("pl.szczeliniak.kitchenassistant"))
            .paths(PathSelectors.any())
            .build()
    }

    private fun apiInfo(): ApiInfo {
        return ApiInfo(
            applicationName,
            applicationDescription,
            applicationVersion,
            null,
            Contact(applicationCreatorName, applicationCreatorUrl, applicationCreatorEmail),
            null,
            null, emptyList()
        )
    }

    private fun apiKey(): ApiKey {
        return ApiKey("JWT", "Authorization", "header")
    }

    private fun securityContext(): SecurityContext? {
        return SecurityContext.builder().securityReferences(defaultAuth()).build()
    }

    private fun defaultAuth(): List<SecurityReference> {
        val authorizationScope = AuthorizationScope("global", "accessEverything")
        val authorizationScopes = arrayOfNulls<AuthorizationScope>(1)
        authorizationScopes[0] = authorizationScope
        return listOf(SecurityReference("JWT", authorizationScopes))
    }


}